### PR TITLE
Implement milestone client for mocks

### PR DESCRIPTION
**Pull request by LePtitDev**

### CODE REVIEW

File: NGitLab.Mock/Clients/MilestoneClient.cs [hunk 1]
```diff
@@ -14,43 +16,119 @@ public MilestoneClient(ClientContext context, int projectId)
             _projectId = projectId;
         }
 
-        public Models.Milestone this[int id] => throw new System.NotImplementedException();
+        public Models.Milestone this[int id]
+        {
+            get
+            {
+                var project = FindProject(_projectId) ?? throw new InvalidOperationException($"Cannot find project with ID {_projectId}");
```
Comment (meziantou) [inline NGitLab.Mock/Clients/MilestoneClient.cs]: Most of the comments from https://github.com/ubisoft/NGitLab/pull/41 applies here too :)

**Review by meziantou — CHANGES_REQUESTED** [review]

File: NGitLab.Mock/Clients/MilestoneClient.cs [hunk 2]
```diff
@@ -14,43 +16,127 @@ public MilestoneClient(ClientContext context, int projectId)
...
         {
-            throw new System.NotImplementedException();
+            using (Context.BeginOperationScope())
+            {
+                var project = GetProject(_projectId, ProjectPermission.Edit);
+                var ms = new Milestone
+                {
+                    Title = milestone.Title,
+                    Description = milestone.Description,
+                    DueDate = string.IsNullOrEmpty(milestone.DueDate) ? DateTimeOffset.Now : DateTimeOffset.Parse(milestone.DueDate),
```
Comment (meziantou) [inline NGitLab.Mock/Clients/MilestoneClient.cs]:
```suggestion
                    DueDate = string.IsNullOrEmpty(milestone.DueDate) ? DateTimeOffset.UtcNow : DateTimeOffset.Parse(milestone.DueDate),
```

File: NGitLab.Mock/Clients/MilestoneClient.cs [hunk 3]
```diff
@@ -14,43 +16,127 @@ public MilestoneClient(ClientContext context, int projectId)
...
+                    StartDate = string.IsNullOrEmpty(milestone.StartDate) ? DateTimeOffset.Now : DateTimeOffset.Parse(milestone.StartDate),
```
Comment (meziantou) [inline NGitLab.Mock/Clients/MilestoneClient.cs]:
```suggestion
                    StartDate = string.IsNullOrEmpty(milestone.StartDate) ? DateTimeOffset.UtcNow : DateTimeOffset.Parse(milestone.StartDate),
```

File: NGitLab.Mock/Clients/MilestoneClient.cs [hunk 4]
```diff
@@ -14,43 +16,127 @@ public MilestoneClient(ClientContext context, int projectId)
...
+                }
+
+                if (!string.IsNullOrEmpty(milestone.DueDate))
+                {
+                    ms.DueDate = DateTimeOffset.Parse(milestone.DueDate);
+                }
+
+                if (!string.IsNullOrEmpty(milestone.StartDate))
+                {
+                    ms.StartDate = DateTimeOffset.Parse(milestone.StartDate);
```
Comment (meziantou) [inline NGitLab.Mock/Clients/MilestoneClient.cs]: You use use an overload that accepts `CultureInfo.InvariantCulture`

File: NGitLab.Mock/Clients/MilestoneClient.cs [hunk 5]
```diff
@@ -14,43 +16,127 @@ public MilestoneClient(ClientContext context, int projectId)
...
+                }
+
+                if (milestone.Description != null)
+                {
+                    ms.Description = milestone.Description;
+                }
+
+                if (!string.IsNullOrEmpty(milestone.DueDate))
+                {
+                    ms.DueDate = DateTimeOffset.Parse(milestone.DueDate);
```
Comment (meziantou) [inline NGitLab.Mock/Clients/MilestoneClient.cs]: You use use an overload that accepts `CultureInfo.InvariantCulture`

**Review by meziantou — APPROVED** [review]

File: NGitLab.Mock.Tests/MilestonesMockTests.cs [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+using System.Linq;
+using NGitLab.Models;
+using NUnit.Framework;
+
+namespace NGitLab.Mock.Tests
+{
+    public class MilestonesMockTests
+    {
+        [Test]
+        public void Test_milestones_from_project_can_be_found()
```
Comment (louis-z) [inline NGitLab.Mock.Tests/MilestonesMockTests.cs]:
```suggestion
        public void Test_milestones_can_be_found_from_project()
```

**Review by louis-z — CHANGES_REQUESTED** [review]

**Review by louis-z — APPROVED** [review]

**Pull request merged**